### PR TITLE
autoadb: init at unstable-2020-06-01

### PR DESCRIPTION
**Pull request by Shawn8901**
###### Description of changes
Autoadb is a small helper which runs commands whenever a device is connected via adb. 
It plays very nicely in combination with scrcpy, which is already in the repository, to automatically start scrcpy when there is a new device attached.

tbh, i was not very sure where to place the import in the `top-level/all-packages.nix`. It makes sense for me to be in the same section like scrcpy. If there is a better place a hint is very welcome. 
If there is something wrong with this PR or something missing, please let me know, 
 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/autoadb/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    owner = "rom1v";
+    repo = pname;
+    rev = "7f8402983603a9854bf618a384f679a17cd85e2d";
+    sha256 = "sha256-9Sv38dCtvbqvxSnRpq+HsIwF/rfLUVZbi0J+mltLres=";
+  };
+
+  cargoSha256 = "sha256-fmNySuOW+2HKIOIqIv/8W41ZXSmq3hbi+11yZBbhW5Q=";
+
+  meta = with lib; {
+    description = "Execute a command whenever a device is adb-connected ";
```
Comment (fabaff) [inline pkgs/misc/autoadb/default.nix:17]:
```suggestion
    description = "Execute a command whenever a device is adb-connected";
```

File: pkgs/misc/autoadb/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = "https://github.com/rom1v/autoadb";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ];
```
Comment (fabaff) [inline pkgs/misc/autoadb/default.nix:20]: New packages need a maintainer, see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

File: pkgs/misc/autoadb/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "autoadb";
+  version = "20200601";
```
Comment (SuperSandro2000) [inline pkgs/misc/autoadb/default.nix:5]:
```suggestion
  version = "unstable-2020-06-01";
```

File: pkgs/misc/autoadb/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  version = "unstable-2020-06-01";
+
+  src = fetchFromGitHub {
+    owner = "rom1v";
+    repo = pname;
+    rev = "7f8402983603a9854bf618a384f679a17cd85e2d";
+    sha256 = "sha256-9Sv38dCtvbqvxSnRpq+HsIwF/rfLUVZbi0J+mltLres=";
+  };
+
+  cargoSha256 = "sha256-fmNySuOW+2HKIOIqIv/8W41ZXSmq3hbi+11yZBbhW5Q=";
```
Comment (SuperSandro2000) [inline pkgs/misc/autoadb/default.nix:14]:
```suggestion
  cargoSha256 = "1gzg1lhq8gp790mrc8fw8dg146k8lg20pnk45m2ssnmdka0826f7";
```

**Pull request merged**